### PR TITLE
Fixed the CI static check issue and bump the golangci-lint version at…

### PR DESCRIPTION
**Pull request by Aisuko**
Signed-off-by: aisuko <urakiny@gmail.com>

1, Who removes the PR template?
2, Bump the version of `golangci-lint` and update the configuration at local
3, Fix the issues of the static-check error and checkout error(We need to do a restrict policy of the code review and merge the code) 
4, Why we replace the CI from `golangci-lint` to more steps CI and make more same steps

### CODE REVIEW

**Comment by ramrodo** [comment]
> I remember you add a codecover to the repo(I do not remember which one), Is it work well?

Yes, I added it to the **meshery** repo by this PR: https://github.com/layer5io/meshery/pull/2258 and here can be seen the current coverage:
https://app.codecov.io/gh/layer5io/meshery

> I believe we should do a strict rule of PR format.

Do you mean to have a template to implement it to all the other repositories?

**Comment by Aisuko** [comment]
> > I remember you add a codecover to the repo(I do not remember which one), Is it work well?
> 
> Yes, I added it to the **meshery** repo by this PR: [layer5io/meshery#2258](https://github.com/layer5io/meshery/pull/2258) and here can be seen the current coverage:
> https://app.codecov.io/gh/layer5io/meshery
> 
> > I believe we should do a strict rule of PR format.
> 
> Do you mean to have a template to implement it to all the other repositories?

Yep, if the coverage or CI process did not succeed, we should not merge the PR.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -9,19 +9,6 @@ on:
     branches:
       - master
```
Comment (leecalcote) [inline .github/workflows/ci.yml:10]: Do these changes bring the Linkerd adapter repo inline with the ci.yml of the other adapter repos (verbatim)?

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,5 @@
 GOPATH = $(shell go env GOPATH)
```
Comment (leecalcote) [inline Makefile:1]: Let's strive for consistency across the repos in this file. Is this the same as other adapter repos?

File: linkerd/custom_operation.go [hunk 3]
```diff
@@ -1,9 +1,11 @@
+// Package linkerd provides custom operation ability for deploying Linkerd2
 package linkerd
 
 import (
 	"github.com/layer5io/meshery-adapter-library/status"
 )
 
+// applyCustomOperation return deployed status or the error message if it exist in deployment process
```
Comment (leecalcote) [inline linkerd/custom_operation.go:8]: Excellent. 🙂

File: linkerd/error.go [hunk 4]
```diff
@@ -21,7 +23,7 @@ var (
 	ErrOpInvalid = errors.NewDefault(errors.ErrOpInvalid, "Invalid operation")
 )
 
-// ErrInstallMesh is the error for install mesh
+// ErrInstallLinkerd is the error for install mesh
```
Comment (leecalcote) [inline linkerd/error.go:26]: Good, good.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Aisuko) [reply .github/workflows/ci.yml:10]: I do not clear what your mean

File: Makefile (hunk 2 above)
Comment (Aisuko) [reply Makefile:1]: I'm afraid not, but this is a good way to get GOPATH by dynamic and do not make an effect on the Makefile feature.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Aisuko) [reply .github/workflows/ci.yml:10]: We use `golangci-lint` includes linter `golint`.  So, we do not need to add another one.

**Comment by Aisuko** [comment]
Close the PR due to after many version of dependencies we updated, some issues already fixed.

**Pull request closed without merging**